pallets/ethereum-xcm/src/lib.rs

```diff
@@ -228,6 +230,39 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Xcm Transact an Ethereum transaction, but allow to force the caller and create address.
+		/// This call should be restricted (callable only by the runtime or governance).
+		/// Weight: Gas limit plus the db reads involving the suspension and proxy checks
+		#[pallet::weight({
+			let without_base_extrinsic_weight = false;
+			<T as pallet_evm::Config>::GasWeightMapping::gas_to_weight({
+				match xcm_transaction {
+					EthereumXcmTransaction::V1(v1_tx) =>  v1_tx.gas_limit.unique_saturated_into(),
+					EthereumXcmTransaction::V2(v2_tx) =>  v2_tx.gas_limit.unique_saturated_into()
+				}
+			}, without_base_extrinsic_weight).saturating_add(T::DbWeight::get().reads(2))
+		})]
+		pub fn force_transact_as(
+			origin: OriginFor<T>,
+			transact_as: H160,
+			xcm_transaction: EthereumXcmTransaction,
+			force_create_address: Option<H160>,
+		) -> DispatchResultWithPostInfo {
+			T::ForceOrigin::ensure_origin(origin)?;
```

Comment:
I guess this `ForceOrigin` will be kind of a wrapper of the existent `XcmEthereumOrigin` right? I wonder if there could be some issues in the ethereum-xcm execution of this function if the origin check is not the same as in the original `transact` call of the pallet.

Reply:
No, it's Root, it add a way for the runtime to submit an ethereum transaction on behalf of any account, actually there is no requirement that it come from XCM

Reply:
> I wonder if there could be some issues in the ethereum-xcm execution of this function if the origin check is not the same as in the original transact call of the pallet.

This call will never de dispached from XCM directly, only from the new palletn it take place on pallet ethereum-xcm to avoid creating yet another new pallet. Actually, the pallet ethereum xcm is not related to XCM on it's own, technically it's just a way to create and execute an ethereum transaction where the caller is the runtime itself and not a user